### DISCUSSION TITLE
[Unknown Mod][Game] Offer to end the game when victory is assured

### DISCUSSION THREAD

**Discussion by LeHerosInconnu** [ihhub/fheroes2#8812] [Ideas]
At a certain point in the game, victory is assured and it's just a matter of completing the scenario.
From this point onwards, the game becomes somewhat boring and of no real interest to certain types of players.

When the conditions for victory or defeat are not specific, the player is offered the possibility of ending and winning the current game via a window displayed at the start of his turn.
The conditions for the window to be displayed are:
- the player must control at least 75% of all towns/castles on the adventure map.
- the strength of all the creatures in the player's heroes' armies and town garrisons must be at least 300% of the same value of all opponents still in play.

The window is presented to the player only once.
If the player decides to continue the game, the window will not be presented again.

To calculate the score and the number of days spent completing the scenario, a third of the days already spent, rounded up, is added to the current number of days already spent to arrive at the day on which the window for ending the game is displayed.

This is an option:
"game: offer to end the game when victory is assured"

This can be implemented directly in fheroes2.

**Comment by ihhub** [comment]
It won't work for maps with Events, especially when we expand their functionality.
